Doc/library/os.path.rst

```diff
@@ -306,17 +306,17 @@ the :mod:`glob` module.)
 
 .. function:: join(path, *paths)
 
-   Join one or more path components intelligently.  The return value is the
+   Join one or more path segments intelligently.  The return value is the
    concatenation of *path* and any members of *\*paths* with exactly one
    directory separator following each non-empty part except the last, meaning
    that the result will only end in a separator if the last part is empty.  If
-   a component is an absolute path, all previous components are thrown away
-   and joining continues from the absolute path component.
+   a segment is an absolute path, all previous segments are ignored
+   and joining continues from the absolute path segment.
```

Comment:
`ntpath` sometimes misuses "absolute" to include a relative rooted path, both in the documentation and the implementation (e.g. `ntpath.isabs()` mishandles "\\spam" as an absolute path). I think it would help to call attention to the meaning of "absolute" that's used here. For example:

```suggestion
   a segment is an absolute path, which on Windows requires both a drive and a
   root, then all previous segments are ignored and joining continues from
   the absolute path segment.
```